x-pack/plugin/downsample/src/main/java/org/elasticsearch/xpack/downsample/LabelFieldProducer.java

```diff
@@ -20,152 +20,139 @@
 import java.util.List;
 
 /**
- * Class that produces values for a label field.
+ * Class that produces the last value of a field for downsampling.
+ * Important note: This class assumes that field values are collected and sorted by descending order by time
  */
-abstract class LabelFieldProducer extends AbstractDownsampleFieldProducer {
+class LabelFieldProducer extends AbstractDownsampleFieldProducer {
```

Comment:
TODO @gmarouli : I will rename this to `LastValueFieldProducer` after the review because github is not rendering correctly the PR and it's hard to  review it.